hw4/hw4.py

```diff
@@ -56,75 +43,59 @@ def lr(df):
     return
 
 
-if __name__ == "__main__":
-    """
-    diabetes = datasets.load_diabetes(as_frame=True)
-    df = diabetes['frame']
-    predictors = ['age','sex','bmi','bp','s1','s2','s3','s4','s5','s6']
-    response = ['target']
-    X = diabetes.data.values
-    y = diabetes.target.values
-    """
-    # iris = datasets.load_iris(as_frame=True)
-    # df = iris['frame']
-    # predictors = ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)','petal width (cm)']
-    # response = ['target']
-    # variable_type(df,predictors,response)
-    # lr(X,y)
-
-
-from sklearn.ensemble import RandomForestClassifier
+def random_forest():
+    import time
 
-# diabetes = datasets.load_diabetes(as_frame=True)
-# df = diabetes['frame']
-X = df.iloc[:, :-1]
-y = df.iloc[:, -1]
-from sklearn.model_selection import train_test_split
+    import numpy as np
+    from sklearn.ensemble import RandomForestClassifier
+    from sklearn.inspection import permutation_importance
+    from sklearn.model_selection import train_test_split
```

Comment:
I don't think it is good practice to have import module statements inside of your functions. I have never seen that before and it can lead to multiple imports into the same script